extensions/HyperVExtension/src/HyperVExtension/Models/VirtualMachineCreation/VMGalleryVMCreationOperation.cs

```diff
@@ -85,67 +89,82 @@ public void Report(IOperationReport value)
+                    await archiveProvider.ExtractArchiveAsync(this, ArchivedFile!, absoluteFilePathForVhd, CancellationTokenSource.Token);
+                    var virtualMachineName = MakeFileNameValid(_userInputParameters.NewEnvironmentName);
 
-                await archiveProvider.ExtractArchiveAsync(this, ArchivedFile!, absoluteFilePathForVhd, CancellationTokenSource.Token);
-                var virtualMachineName = MakeFileNameValid(_userInputParameters.NewEnvironmentName);
+                    // Use the Hyper-V manager to create the VM.
+                    UpdateProgress(_stringResource.GetLocalized("CreationInProgress", virtualMachineName));
+                    var creationParameters = new VirtualMachineCreationParameters(
+                        _userInputParameters.NewEnvironmentName,
+                        GetVirtualMachineProcessorCount(),
+                        absoluteFilePathForVhd,
+                        Image.Config.SecureBoot,
+                        Image.Config.EnhancedSessionTransportType);
 
-                // Use the Hyper-V manager to create the VM.
-                UpdateProgress(_stringResource.GetLocalized("CreationInProgress", virtualMachineName));
-                var creationParameters = new VirtualMachineCreationParameters(
-                    _userInputParameters.NewEnvironmentName,
-                    GetVirtualMachineProcessorCount(),
-                    absoluteFilePathForVhd,
-                    Image.Config.SecureBoot,
-                    Image.Config.EnhancedSessionTransportType);
+                    return new CreateComputeSystemResult(_hyperVManager.CreateVirtualMachineFromGallery(creationParameters));
+                }
+                catch (Exception ex)
+                {
+                    if (creationAttempt == maxAttempts)
+                    {
+                        _log.Error(ex, "Operation to create compute system failed on the last attempt");
+                        return new CreateComputeSystemResult(ex, ex.Message, ex.Message);
+                    }
+                    else
+                    {
+                        _log.Error(ex, "Operation to create compute system failed on the first attempt, retrying.");
+                    }
+                }
 
-                ComputeSystemResult = new CreateComputeSystemResult(_hyperVManager.CreateVirtualMachineFromGallery(creationParameters));
-            }
-            catch (Exception ex)
-            {
-                _log.Error(ex, "Operation to create compute system failed");
-                ComputeSystemResult = new CreateComputeSystemResult(ex, ex.Message, ex.Message);
+                creationAttempt++;
             }
 
-            IsOperationCompleted = true;
-            IsOperationInProgress = false;
+            // We shouldn't get here since we should either complete successfully or throw an error and
+            // send that error message back to Dev Home on the second try if it fails..
```

Comment:
Comment assumes only 2 attempts.  Remove references to the number of attempts because the number might change in the future.

Reply:
updated thanks